pkgs/development/interpreters/lua-5/hooks/setup-hook.sh

```diff
@@ -16,13 +16,21 @@ addToLuaSearchPathWithCustomDelimiter() {
   # delete longest match starting from the lua placeholder '?'
   local topDir="${absPattern%%\?*}"
 
-  # export only if the folder exists else LUA_PATH/LUA_CPATH grow too large
-  if [[ ! -d "$topDir" ]]; then return; fi
-
   # export only if we haven't already got this dir in the search path
   if [[ ${!varName-} == *"$absPattern"* ]]; then return; fi
 
-  export "${varName}=${!varName:+${!varName};}${absPattern}"
+  # export only if the folder contains lua files
+  shopt -s globstar
+  for _file in "$topDir"/**.lua; do
+    export "${varName}=${!varName:+${!varName};}${absPattern}"
+    return;
+  done
```

Comment:
using `**` here seems like a bad idea as long `$topDir` could easily be a derivation's root

Reply:
why is it a bad idea ? the purpose of that code is to filter out derivations that dont add anything to lua else you end up with big LUA_PATH.

Reply:
the advantage of the glob is that we can remove the `LuaPathSearchPaths` repetition you mentioned at the cost of including derivations that may not contain lua files readable from the interpreter LUA_PATH. It's an ok tradeoff I would say

Reply:
apon further review of the code, i realized this isn't actually the problem, but instead the problem was with unprefixed paths being added to LUA_PATH_PATTERN, but since that has been taken out, this issue is now resolved.